CRM/Utils/Geocode.php

```diff
@@ -34,23 +34,18 @@
 /**
  * Class CRM_Utils_Geocode
  */
-class CRM_Utils_Geocode {
+abstract class CRM_Utils_Geocode {
 
   /**
-   * @return string|''
-   *   Class name, or empty.
+   * Updates the passed address with additional information (e.g.,
+   * latitude/longitude, probably by querying an external service).
+   *
+   * @param array $values
+   * @param bool $stateName
+   *
+   * @return bool
+   *   TRUE if we modified the address, FALSE otherwise
    */
-  public static function getProviderClass() {
```

Comment:
This function IS currently called by virtue of  in CRM_Core_Config    
```
 // "callback" properties are generated on-demand by calling a function.
      'geocodeMethod' => array('callback', 'CRM_Utils_Geocode', 'getProviderClass'),
```

Reply:
Wow. I didn't know we had callbacks in CRM_Core_Config. I bet the `cv` command in my earlier comment didn't work because I had already removed the callback.

Updating that reference as follows should do the trick:
`'geocodeMethod' => array('callback', 'CRM_Utils_GeocodeProviderFactory', 'getUsableClassName'),`

I'm not sure whether or not we should be deprecating use of config callbacks in general with the advent of the `Civi` interface. Either way, it looks like there are enough uses of it that we ought to have a closer look to make sure we aren't breaking anything.